notebooks/224-stable-diffusion-text-to-image/224-stable-diffusion-text-to-image.ipynb

```diff
@@ -0,0 +1,1209 @@
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Convert models to OpenVINO Intermediate representation format\n",
+    "\n",
+    "OpenVINO supports PyTorch\\* through export to the ONNX\\* format. We will use `torch.onnx.export` function for obtaining ONNX, \n",
+    "you can find more info about it in [PyTorch documentation](https://pytorch.org/docs/stable/onnx.html). \n",
+    "We need provide model object, input data for model tracing and path for model saving. \n",
+    "Optioanally, we can provide target onnx opset for conversion and other parameters specified in documentation (e.g. input and output names or dynamic shapes).\n",
+    "\n",
+    "While ONNX models are directly supported by OpenVINO™, it can be useful to convert them to IR format to take advantage of advanced OpenVINO optimization tools and features. `mo.convert` function can be used for converting model using OpenVINO Model Optimizer capabilities. It returns of instance OpenVINO Model class, which is ready to use in python interface and can be serialized to IR for next usages.\n",
+    "\n",
+    "The model consists from three important parts:\n",
+    "* Text Encoder for creation condition to generate image from text prompt\n",
+    "* Unet for step by step denoising latent image representation\n",
+    "* Autoencoder (VAE) for decoding latent space to image\n",
+    "\n",
+    "Let's convert each part"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Text Encoder\n",
+    "\n",
+    "The text-encoder is responsible for transforming the input prompt, e.g. \"a photo of an astronaut riding a horse\" into an embedding space that can be understood by the U-Net. It is usually a simple transformer-based encoder that maps a sequence of input tokens to a sequence of latent text embeddings.\n",
+    "\n",
+    "Input of text encoder is tensor `input_ids` which contains indexes of tokens from text processed by tokenizer and padded to maximum length accepted by model.\n",
+    "Output of model is 2 tensors: `last_hidden_state` - hidden state from the last MultiHeadAttention layer in model and `pooler_out` - Pooled output for whole model hidden states. We will use `opset_version=14`, because model contains `triu` operation, supported in ONNX only starting from this opset."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 52,
+   "metadata": {
+    "colab": {
+     "base_uri": "https://localhost:8080/"
+    },
+    "id": "U1cQFbVS0Ugf",
+    "outputId": "9b3b825f-2adc-4d90-cae0-74890d368eab"
+   },
+   "outputs": [],
+   "source": [
+    "from pathlib import Path\n",
+    "import torch\n",
+    "from openvino.tools import mo\n",
+    "from openvino.runtime import serialize\n",
+    "\n",
+    "TEXT_ENCODER_ONNX_PATH = Path('text_encoder.onnx')\n",
+    "TEXT_ENCODER_OV_PATH = TEXT_ENCODER_ONNX_PATH.with_suffix('.xml')\n",
+    "\n",
+    "\n",
+    "def convert_encoder(pipe, onnx_path, ov_path):\n",
```

Comment:
Please use doc string and type hints in order to create a better understanding of the functions:

Similar to this image for all functions, classes and methods.

![image](https://user-images.githubusercontent.com/10940214/200874931-787e1f36-3b9c-4e29-a93d-f44d802ad06f.png)

Reply:
added